File: Practice/AllDomains/Tutorials/30DaysOfCode/Day8DictionariesAndMaps.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution 
{
	/*	author:@shivkrthakur  */
	static void Main(String[] args) 
	{
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int x = Convert.ToInt32(Console.ReadLine().Trim());
        
        Dictionary<string, string> phoneBook = new Dictionary<string, string>(x);
        for(int i = 0; i < x; i++)
        {
            string nameAndPhone = Console.ReadLine();
            string name = nameAndPhone.Substring(0, nameAndPhone.IndexOf(' '));
            string phone = nameAndPhone.Substring(nameAndPhone.IndexOf(' ')+1);
            
            phoneBook[name] = phone;
        }
        
        for(int i = 0; i < x; i++)
        {
            string name = Console.ReadLine();
            if(phoneBook.ContainsKey(name))
            {
                Console.WriteLine("{0}={1}", name, phoneBook[name]);
            }
            else 
            {
                Console.WriteLine("Not found");    
            }
        }
    }
}

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day8DictionariesandMaps.js
/*	author:@shivkrthakur  */
var inputArr = [];
var inputIndex = 0;

function readLine(){
    return inputArr[inputIndex++];
}
function processData(input) {
    //Enter your code here
    inputArr = input.split('\n');
    //console.log(inputArr);
    var c = parseInt(readLine());
    var dict = {};
    for(var i = 0; i < c; i++){
        var keyValue = readLine().split(' ');
        dict[keyValue[0]] = keyValue[1];
    }
    var inputString;
    do{
        if(inputString){
            if(dict[inputString]) console.log(inputString + "=" + dict[inputString]);
            else console.log("Not found");
        }
        inputString = readLine();
    }while(inputString)
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
